### PR TITLE
Upgrade to latest react-map-gl

### LINKED ISSUE #476: Update to latest `react-map-gl`
https://visgl.github.io/react-map-gl/

### PR DESCRIPTION
**Pull request by 0x41head**
closes #476 
- update to `map` from `ReactMapGL`
- removed `LinearInterpolator`
- Fixed `WebMercatorViewport` import

### CODE REVIEW

**Comment by adeebshihadeh** [comment]
Getting a bunch of errors when trying to run:
image

**Comment by 0x41head** [comment]
Made some changes that should theoretically fix this issue. If for some reason it still persists, please inform me about your browser and browser version, so that I can test it there

**Comment by adeebshihadeh** [comment]
Still doesn't work. Wonder why the tests pass?
```
Google Chrome	120.0.6099.129 (Official Build) (64-bit) 
Revision	204016af461e03117aa3858b51ba4534c7cfda81-refs/branch-heads/6099_110@{#3}
OS	Linux
JavaScript	V8 12.0.267.10
User Agent	Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/120.0.0.0 Safari/537.36
```

**Comment by 0x41head** [comment]
Cannot replicate the error, even on the same browser version :/ 
Made a few changes that could help.
Is there anything specific you are doing to cause this issue or is it directly on boot ?
Do you have any extension that could potentially cause this issue ? Maybe try once on incognito ?

**Comment by 0x41head** [comment]
Nevermind, I was able to replicate the issue somehow. Working on a fix rn

**Comment by 0x41head** [comment]
Ready for review.

**Comment by adeebshihadeh** [comment]
Works a little more but immediately hit a visual bug. All these things should be caught before the PR is marked ready for review. Try to break it before marking as ready for review again.

image

**Comment by 0x41head** [comment]
Apologies for the regular pings. 
I think a lot of these oversights are happening because I am using the demo account, while you are using an actual comma device. In order to actually verify a lot of these issues, I have to remove the conditional rendering happening at multiple places. For example, the `hasNav` variable only gets defined if you actually have a prime device with navigation feature
On a side note, we should probably try to integrate these into the unit test as well for easier development in the future.

**Comment by adeebshihadeh** [comment]
Can you rebase to get the new preview deployment? Then you can share the link on our Discord and have some people stress test it.

File: src/index.css [hunk 1]
```diff
@@ -44,6 +44,12 @@ a:focus {
   outline: none;
   box-shadow: none;
 }
+.mapboxgl-ctrl-top-right {
+  position: absolute;
+   width: 100%;
```
Comment (adeebshihadeh) [inline src/index.css]: wrong indentation

File: src/components/Navigation/index.jsx [hunk 2]
```diff
@@ -1242,6 +1238,7 @@ class Navigation extends Component {
         />
         { search && !searchSelect && !searchLooking && (
         <>
+          {console.log(search)}
```
Comment (adeebshihadeh) [inline src/components/Navigation/index.jsx]: remove this

File: package.json [hunk 3]
```diff
@@ -29,6 +29,8 @@
     "@mapbox/mapbox-sdk": "^0.15.3",
     "@material-ui/core": "^1.5.1",
     "@material-ui/icons": "^1.1.1",
+    "@math.gl/core": "^4.0.1",
+    "@math.gl/web-mercator": "^4.0.1",
```
Comment (adeebshihadeh) [inline package.json:33]: why do we need new dependencies?

File: src/components/DriveMap/index.jsx [hunk 4]
```diff
@@ -158,6 +144,22 @@ class DriveMap extends Component {
   onViewportChange(viewport) {
     this.setState({ viewport });
   }
+  
+  onMove({viewState,type}){
+    this.setState({ viewport:viewState });
+
+    if (type=="move") {
```
Comment (adeebshihadeh) [inline src/components/DriveMap/index.jsx]: missing whitespace

File: src/components/DriveMap/index.jsx [hunk 5]
```diff
@@ -158,6 +144,22 @@ class DriveMap extends Component {
   onViewportChange(viewport) {
     this.setState({ viewport });
   }
+  
+  onMove({viewState,type}){
+    this.setState({ viewport:viewState });
```
Comment (adeebshihadeh) [inline src/components/DriveMap/index.jsx]: and here too

File: src/components/Navigation/index.jsx [hunk 6]
```diff
@@ -837,6 +841,7 @@ class Navigation extends Component {
   }
 
   onResize(windowWidth) {
+    console.log('windowWidth',windowWidth)
```
Comment (adeebshihadeh) [inline src/components/Navigation/index.jsx]: remove this

File: src/components/Navigation/index.jsx [hunk 7]
```diff
@@ -903,10 +908,10 @@ class Navigation extends Component {
   }
 
   researchArea() {
-    const { viewport, windowWidth } = this.state;
+    const { windowWidth } = this.state;
     const searchInput = this.searchInputRef.current;
-
-    forwardLookup(searchInput.value, null, viewport).then((features) => {
+    const bbox =this.mapRef.current.getBounds()
```
Comment (adeebshihadeh) [inline src/components/Navigation/index.jsx]: missing whitespace

**Review by adeebshihadeh — CHANGES_REQUESTED** [review]
Can you clean this up? It's got debug stuff leftover and a bunch of inconsistent style

File: package.json (hunk 3 above)
Comment (0x41head) [reply package.json:33]: from react-map-gl upgrade docs:

`viewport-mercator-project` (an alias of `@math.gl/web-mercator`) is no longer a dependency. You can still install the library on the side as a utility for viewport-related math, but it's no longer required.

Comment (0x41head) [reply package.json:33]: [Reference](https://visgl.github.io/react-map-gl/docs/upgrade-guide)

**Comment by adeebshihadeh** [comment]
I'm closing out this bounty, since we're shifting our focus over to [new-connect](https://github.com/commaai/new-connect).

The work is done though, so you're good to collect the bounty. Instructions here: https://github.com/commaai/openpilot/blob/master/docs/BOUNTIES.md

**Pull request closed without merging**